frontend/packages/shared/src/providers/PreviewConnectionContext.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { createContext, ReactNode, useContext, useEffect, useState } from 'react';
+import { previewSignalRHubSubPath } from 'app-shared/api-paths';
+import { HubConnection, HubConnectionBuilder, LogLevel } from "@microsoft/signalr";
+
+const PreviewConnectionContext = createContext<HubConnection>(null);
+
+interface PreviewConnectionContextProviderProps {
+  children: ReactNode;
+}
+
+export const PreviewConnectionContextProvider = ({ children }: PreviewConnectionContextProviderProps ) => {
+  const [connection, setConnection] = useState<HubConnection>(null);
+
+  useEffect(() => {
+    const newConnection = new HubConnectionBuilder().withUrl(previewSignalRHubSubPath()).configureLogging(LogLevel.Information).build();
+    setConnection(newConnection);
+  }, []);
+
+  useEffect(() => {
+    async function start() {
+      await connection.start().catch(e => console.log('Connection failed: ', e));
+    }
+
+    if (connection) {
+      start();
+    }
```

Comment:
Dette kan forenkles litt:
```suggestion
    if (connection) await connection.start().catch(e => console.log('Connection failed:', e));
```

Reply:
Går det med async useEffect? 🤔

Reply:
Nei, godt poeng, det vil jo gjøre at det returneres en `Promise`, så det blir jo faktisk ikke gyldig. Hva skjer hvis du fjerner `await` fullstendig? Det burde gi samme resultat som tidligere. Virker ikke det heller, lar du det bare stå som det var 🙂